### DISCUSSION TITLE
Newby Question: How to enable HTTPS with Cloudflare Origin Certificate

### DISCUSSION THREAD

**Discussion by l3m0n42** [dani-garcia/vaultwarden#3525] [Q&A]
I am running the Valutwarden through a Cloudflare tunnel. I initially set the protocol on the Cloudflare tunnel with HTTP and was able to make configurations within the internal network that I am hosting the Vaultwarden server on. However, when I try to make a new account from an external network outside of where I am hosting the service, I am unable to make configurations due to the connection not being secure. I then tried to set the protocol as HTTPS and am now having a bad gateway issue. I have an origin server cert through Cloudflare which gave me the certificate and private key however I am not sure how to use these with the vault warden container. Basically: Is there a way to enable HTTPS with the Cloudflare cert and private key? Sorry for the bothersome question and thanks for any help.

**Comment by Masgalor** [comment]
Have you already read the Wiki article for this topic?
https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS

If you followed the steps described there, please tell us how exactly you enabled HTTPS.

> now having a bad gateway issue

Sounds like your service is now bound to port 443 but your proxy still connects to port 80.

**Comment by Raito00** [comment]
I use just Cloudflare DNS with proxy on - to hide server IP
My instalation:
Nginx proxy manager - redirect to Vaultwarden docker and issue ssl cert for domain name
Vaultwarden docker with Caddy - (Caddy) to enable SSL (HTTPS)

**Reply by giuse17ita** [reply]
im noob, and this is my question:
if i have my server with a https reverse proxy, behind cloudflare cdn. how data travel?
i suppose that: 
client encrypt and send to cloudflare 
then 
cloudflare decrypt, reencrypt and send to server.

and vice versa

its correct? how it works?

**Reply by kpfleming** [reply]
Yes that is correct.

**Reply by giuse17ita** [reply]
so, cloudflare what can see/decrypt?

**Reply by kpfleming** [reply]
If you use Cloudflare in front of your Vaultwarden server, then Cloudflare can see all of the traffic between the server and its clients. The secret parts of that traffic are encrypted with the user's master passwords, so it is not exposed to Cloudflare.
